drivers/video/video_esp32_dvp.c

```diff
@@ -389,6 +389,34 @@ static int video_esp32_init(const struct device *dev)
 	return 0;
 }
 
+int video_esp32_set_selection(const struct device *dev, struct video_selection *sel)
+{
+	struct video_esp32_data *data = dev->data;
+	const struct video_esp32_config *cfg = dev->config;
+	int ret;
+
+	ret = video_set_selection(cfg->source_dev, sel);
+	if (ret) {
+		LOG_ERR("Failed to set selection on source device");
+		return ret;
+	}
+
+	data->video_format.width = sel->rect.width;
+	data->video_format.height = sel->rect.height;
+	data->video_format.pitch =
+		data->video_format.width * video_bits_per_pixel(data->video_format.pixelformat) /
+		BITS_PER_BYTE;
```

Comment:
Can `video_esp32_get_fmt(dev, &data->video_format)` be called instead?
This would ask the source device, expected to have this computation done already.